.github/workflows/generate.yml

```diff
@@ -0,0 +1,49 @@
+name: Code Generation
+
+on:
+  push:
+    branches: ["main"]
+
+  workflow_dispatch:
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: frenck/action-setup-yq@v1
+      - name: Downgrade OAS
+        run: bash scripts/downgrade-oas-version.sh
+  build:
+    needs: [ setup ]
+    runs-on: ubuntu-latest
+    strategy:
+            matrix:
+                target:
+                    - go-gin-server
+                    - python-flask 
+                cx-version:
+                    # downgradePath represents the spec file path as created as 
+                    # part of scripts/downgrade-oas-version.sh in a previous job 
+                    - {semantic: "5.0", downgradePath: "5dot0"}
+                    - {semantic: "4.6", downgradePath: "4dot6"}
+
+    steps:
+      - run: echo "Generate Code"
+      - name: Generate Server ${{ specPath }} 
+        uses: openapi-generators/openapitools-generator-action@v1
+        with:
+          generator: ${{ matrix.target }}
+          openapi-file: ./.backward-compatible-versions/corex-${{matrix.cx-version.downgradePath}}-304.yaml
```

Comment:
I'm a little confused. Are these _language-specific_ server files that will be added to the `.backward-compatible-versions` directory going forward?

Reply:
Nope, so the OAS generator does not work with version 3.1 so we have to downgrade the spec files to 3.04.

So first we create a folder and copy ally the spec versions as version 3.04 to that folder.

Then we use each spec version to generate each target here

we dont need to commit the backwards compatible versions because they are created within the gh action machine only for the use of creating generated code